crates/runtime/src/datafusion/cluster/common/datafusion_scheduler_ext.rs

```diff
@@ -0,0 +1,36 @@
+use crate::datafusion::DataFusion;
+use ballista_core::serde::scheduler::ExecutorMetadata;
+use ballista_scheduler::state::SchedulerState;
+use datafusion::common::{DataFusionError, Result};
+use datafusion_proto::logical_plan::AsLogicalPlan;
+use datafusion_proto::physical_plan::AsExecutionPlan;
+use datafusion_proto::protobuf::{LogicalPlanNode, PhysicalPlanNode};
+use std::sync::Arc;
+use std::time::Duration;
+use tokio::runtime::Handle;
+use tokio::task;
+
+/// Some convenience methods for the `DataFusion` for accessing the scheduler state in clustered mode
+pub trait DataFusionSchedulerExtensions<T: 'static + AsLogicalPlan, U: 'static + AsExecutionPlan> {
```

Comment:
The scheduler state exposes things like the executor fleet size, which may be useful in making better execution plans during the planning stage.